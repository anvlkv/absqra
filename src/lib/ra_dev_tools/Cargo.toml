[package]
name = "ra_dev_tools"
version = "0.1.0"
authors = ["a.nvlkv <a.nvlkv@protonmail.ch>"]
edition = "2018"
build = "build.rs"

[build-dependencies]
build-deps = "0.1.4"
# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html

[dependencies]
insta = "0.16.0"
regex = "1.3.9"
example_files_macro = {path="./src/lib/example_files_macro"}
files_iterator = {path="./src/lib/files_iterator"}
