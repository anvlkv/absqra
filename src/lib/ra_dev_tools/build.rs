extern crate build_deps;

fn main() {
    // Enumerate files in sub-folder "examples/*", being relevant for the test-generation (as example)
    // If function returns with error, exit with error message.
    build_deps::rerun_if_changed_paths( "../../../examples/*/*" ).unwrap();

    // Adding the parent directory "examples" to the watch-list will capture new-files being added
    build_deps::rerun_if_changed_paths( "../../../examples/*" ).unwrap();
}