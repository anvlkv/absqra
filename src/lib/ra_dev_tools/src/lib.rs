pub extern crate insta;
pub use files_iterator::for_each_ra_example_file;

#[allow(unused_attributes)]
#[macro_use] pub use example_files_macro::make_example_tests;

#[cfg(test)]
mod test {
    use super::make_example_tests;

    #[make_example_tests]
    #[test]
    fn it_works_with(content: String, file_name: String) {
        println!("didn't panic with content: \n[{}] of file_name: {}", content, file_name);
    }
}