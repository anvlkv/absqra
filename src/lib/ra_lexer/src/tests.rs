use super::*;
use ra_dev_tools::insta::assert_json_snapshot;
use ra_dev_tools::make_example_tests;

#[make_example_tests]
#[test]
fn it_should_match_snapshots(contents: String, file_name: String) {
    let tokens: Vec<RaToken> = tokenize(&contents).map(|r| r.unwrap()).collect();
    assert_json_snapshot!(file_name, tokens)
}