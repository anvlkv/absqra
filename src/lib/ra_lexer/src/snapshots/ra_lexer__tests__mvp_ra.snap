---
source: src/lib/ra_lexer/src/tests.rs
expression: tokens
---
[
  {
    "kind": {
      "Comment": [
        " this file is my todo list and also should render like one",
        false
      ]
    },
    "position": [
      [
        1,
        1
      ],
      [
        2,
        0
      ]
    ],
    "level": 1
  },
  {
    "kind": {
      "Identifier": "parseBlocks"
    },
    "position": [
      [
        3,
        1
      ],
      [
        4,
        4
      ]
    ],
    "level": 1
  },
  {
    "kind": "Greater",
    "position": [
      [
        4,
        5
      ],
      [
        4,
        6
      ]
    ],
    "level": 2
  },
  {
    "kind": "Exclamation",
    "position": [
      [
        5,
        9
      ],
      [
        5,
        10
      ]
    ],
    "level": 2
  },
  {
    "kind": {
      "Identifier": "choice"
    },
    "position": [
      [
        5,
        10
      ],
      [
        5,
        16
      ]
    ],
    "level": 2
  },
  {
    "kind": "Colon",
    "position": [
      [
        5,
        16
      ],
      [
        5,
        17
      ]
    ],
    "level": 2
  },
  {
    "kind": {
      "Identifier": "multiple"
    },
    "position": [
      [
        5,
        17
      ],
      [
        6,
        8
      ]
    ],
    "level": 2
  },
  {
    "kind": "Exclamation",
    "position": [
      [
        6,
        9
      ],
      [
        6,
        10
      ]
    ],
    "level": 2
  },
  {
    "kind": {
      "Identifier": "options"
    },
    "position": [
      [
        6,
        10
      ],
      [
        6,
        17
      ]
    ],
    "level": 2
  },
  {
    "kind": "Colon",
    "position": [
      [
        6,
        17
      ],
      [
        6,
        18
      ]
    ],
    "level": 2
  },
  {
    "kind": {
      "ContentBlock": "parse blocks"
    },
    "position": [
      [
        7,
        13
      ],
      [
        8,
        12
      ]
    ],
    "level": 3
  },
  {
    "kind": {
      "ContentBlock": "and funny chars too ðŸ˜ "
    },
    "position": [
      [
        8,
        13
      ],
      [
        9,
        12
      ]
    ],
    "level": 3
  },
  {
    "kind": {
      "ContentBlock": "â‚°"
    },
    "position": [
      [
        9,
        13
      ],
      [
        10,
        12
      ]
    ],
    "level": 3
  },
  {
    "kind": {
      "ContentBlock": "Ï¿"
    },
    "position": [
      [
        10,
        13
      ],
      [
        11,
        0
      ]
    ],
    "level": 3
  },
  {
    "kind": {
      "Identifier": "parseDifferentTypesOfBlocks"
    },
    "position": [
      [
        12,
        1
      ],
      [
        13,
        4
      ]
    ],
    "level": 1
  },
  {
    "kind": "Greater",
    "position": [
      [
        13,
        5
      ],
      [
        13,
        6
      ]
    ],
    "level": 1
  },
  {
    "kind": "Exclamation",
    "position": [
      [
        14,
        9
      ],
      [
        14,
        10
      ]
    ],
    "level": 2
  },
  {
    "kind": {
      "Identifier": "choice"
    },
    "position": [
      [
        14,
        10
      ],
      [
        14,
        16
      ]
    ],
    "level": 2
  },
  {
    "kind": "Colon",
    "position": [
      [
        14,
        16
      ],
      [
        14,
        17
      ]
    ],
    "level": 2
  },
  {
    "kind": {
      "Identifier": "multiple"
    },
    "position": [
      [
        14,
        17
      ],
      [
        15,
        8
      ]
    ],
    "level": 2
  },
  {
    "kind": "Exclamation",
    "position": [
      [
        15,
        9
      ],
      [
        15,
        10
      ]
    ],
    "level": 2
  },
  {
    "kind": {
      "Identifier": "options"
    },
    "position": [
      [
        15,
        10
      ],
      [
        15,
        17
      ]
    ],
    "level": 2
  },
  {
    "kind": "Colon",
    "position": [
      [
        15,
        17
      ],
      [
        15,
        18
      ]
    ],
    "level": 2
  },
  {
    "kind": {
      "ContentBlock": "comments"
    },
    "position": [
      [
        16,
        13
      ],
      [
        17,
        12
      ]
    ],
    "level": 3
  },
  {
    "kind": {
      "ContentBlock": "var declaration"
    },
    "position": [
      [
        17,
        13
      ],
      [
        18,
        12
      ]
    ],
    "level": 3
  },
  {
    "kind": {
      "ContentBlock": "rule declaration"
    },
    "position": [
      [
        18,
        13
      ],
      [
        19,
        12
      ]
    ],
    "level": 3
  },
  {
    "kind": {
      "ContentBlock": "applied rule"
    },
    "position": [
      [
        19,
        13
      ],
      [
        20,
        12
      ]
    ],
    "level": 3
  },
  {
    "kind": {
      "ContentBlock": "input block"
    },
    "position": [
      [
        20,
        13
      ],
      [
        21,
        12
      ]
    ],
    "level": 3
  },
  {
    "kind": {
      "ContentBlock": "context switches"
    },
    "position": [
      [
        21,
        13
      ],
      [
        22,
        12
      ]
    ],
    "level": 3
  },
  {
    "kind": {
      "ContentBlock": "content block"
    },
    "position": [
      [
        22,
        13
      ],
      [
        23,
        0
      ]
    ],
    "level": 3
  },
  {
    "kind": {
      "Identifier": "generateRuntime"
    },
    "position": [
      [
        24,
        1
      ],
      [
        25,
        4
      ]
    ],
    "level": 1
  },
  {
    "kind": "Greater",
    "position": [
      [
        25,
        5
      ],
      [
        25,
        6
      ]
    ],
    "level": 1
  },
  {
    "kind": "Exclamation",
    "position": [
      [
        26,
        9
      ],
      [
        26,
        10
      ]
    ],
    "level": 2
  },
  {
    "kind": {
      "Identifier": "choice"
    },
    "position": [
      [
        26,
        10
      ],
      [
        26,
        16
      ]
    ],
    "level": 2
  },
  {
    "kind": "Colon",
    "position": [
      [
        26,
        16
      ],
      [
        26,
        17
      ]
    ],
    "level": 2
  },
  {
    "kind": {
      "Identifier": "multiple"
    },
    "position": [
      [
        26,
        17
      ],
      [
        27,
        8
      ]
    ],
    "level": 2
  },
  {
    "kind": "Exclamation",
    "position": [
      [
        27,
        9
      ],
      [
        27,
        10
      ]
    ],
    "level": 2
  },
  {
    "kind": {
      "Identifier": "options"
    },
    "position": [
      [
        27,
        10
      ],
      [
        27,
        17
      ]
    ],
    "level": 2
  },
  {
    "kind": "Colon",
    "position": [
      [
        27,
        17
      ],
      [
        27,
        18
      ]
    ],
    "level": 2
  },
  {
    "kind": {
      "ContentBlock": "render content"
    },
    "position": [
      [
        28,
        13
      ],
      [
        29,
        12
      ]
    ],
    "level": 3
  },
  {
    "kind": {
      "ContentBlock": "render inputs"
    },
    "position": [
      [
        29,
        13
      ],
      [
        30,
        12
      ]
    ],
    "level": 3
  },
  {
    "kind": {
      "ContentBlock": "group inputs as long as there no dependency on previous step"
    },
    "position": [
      [
        30,
        13
      ],
      [
        31,
        12
      ]
    ],
    "level": 3
  },
  {
    "kind": {
      "ContentBlock": "execute logic"
    },
    "position": [
      [
        31,
        13
      ],
      [
        32,
        12
      ]
    ],
    "level": 3
  },
  {
    "kind": {
      "ContentBlock": "apply rules"
    },
    "position": [
      [
        32,
        13
      ],
      [
        33,
        0
      ]
    ],
    "level": 3
  }
]
